doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown

```diff
@@ -0,0 +1,802 @@
+
+@code{.cpp}
+cv::Mat outputImage = inputImage.clone();
+cv::aruco::drawDetectedMarkers(outputImage, markerCorners, markerIds);
+@endcode
+
+- `outputImage ` is the input/output image where the markers will be drawn (it will normally be the same as the image where the markers were detected).
+- `markerCorners` and `markerIds` are the structures of the detected markers returned by the `cv::aruco::ArucoDetector::detectMarkers()` function.
+
+![Image with detected markers](images/singlemarkersdetection.jpg)
+
+Note that this function is only provided for visualization and its use can be omitted.
+
+With these two functions we can create a basic marker detection loop to detect markers from our
+camera:
+
+@snippet samples/cpp/tutorial_code/objectDetection/detect_markers.cpp aruco_detect_markers
+
+Note that some of the optional parameters have been omitted, like the detection parameter object and the
+output vector of rejected candidates.
+
+A full working example is included in the `detect_markers.cpp` inside the `samples/cpp/tutorial_code/objectDetection/`.
+
+Note: The samples now take input from the command line using cv::CommandLineParser. For this file the example parameters will look like
+@code{.cpp}
+-v=/path_to_opencv/opencv/doc/tutorials/objdetect/aruco_detection/images/singlemarkersoriginal.jpg -d=10
+@endcode
+Parameters for `detect_markers.cpp`:
+@snippet samples/cpp/tutorial_code/objectDetection/detect_markers.cpp aruco_detect_markers_keys
+
+
+Pose Estimation
+------
+
+The next thing you'll probably want to do after detecting the markers is to use them to get the camera pose.
+
+To perform camera pose estimation, you need to know your camera's calibration parameters. These are
+the camera matrix and distortion coefficients. If you do not know how to calibrate your camera, you can
+take a look at the `calibrateCamera()` function and the Calibration tutorial of OpenCV. You can also calibrate your camera using the aruco module
+as explained in the **Calibration with ArUco and ChArUco** tutorial. Note that this only needs to be done once unless the
+camera optics are modified (for instance changing its focus).
+
+As a result of the calibration, you get a camera matrix: a matrix of 3x3 elements with the
+focal distances and the camera center coordinates (a.k.a intrinsic parameters), and the distortion
+coefficients: a vector of 5 or more elements that models the distortion produced by your camera.
+
+When you estimate the pose with ArUco markers, you can estimate the pose of each marker individually.
+If you want to estimate one pose from a set of markers, use ArUco Boards (see the **Detection of ArUco
+Boards** tutorial). Using ArUco boards instead of single markers allows some markers to be occluded.
+
+The camera pose relative to the marker is a 3d transformation from the marker coordinate system to the
+camera coordinate system. It is specified by rotation and translation vectors (see `cv::solvePnP()` function for more
```

Comment:
It is specified by rotation and translation vectors. OpenCV provides cv::solvePnP function to do that.

Reply:
fixed